HW2-kata3.py

```diff
@@ -0,0 +1,56 @@
+""" 5 kyu - Pick peaks https://www.codewars.com/kata/5279f6fe5ab7f447890006a7
+
+In this kata, you will write a function that returns the positions and the
+values of the "peaks" (or local maxima) of a numeric array.
+
+For example, the array arr = [0, 1, 2, 5, 1, 0] has a peak at position 3
+with a value of 5 (since arr[3] equals 5).
+
+The output will be returned as an object with two properties: pos and peaks.
+Both of these properties should be arrays. If there is no peak in the given
+array, then the output should be {pos: [], peaks: []}.
+
+Example: pickPeaks([3, 2, 3, 6, 4, 1, 2, 3, 2, 1, 2, 3]) should return {pos:
+[3, 7], peaks: [6, 3]} (or equivalent in other languages)
+
+All input arrays will be valid integer arrays (although it could still be
+empty), so you won't need to validate the input.
+
+The first and last elements of the array will not be considered as peaks (in
+the context of a mathematical function, we don't know what is after and
+before and therefore, we don't know if it is a peak or not).
+
+Also, beware of plateaus !!! [1, 2, 2, 2, 1] has a peak while [1, 2, 2, 2,
+3] does not. In case of a plateau-peak, please only return the position and
+value of the beginning of the plateau. For example: pickPeaks([1, 2, 2, 2,
+1]) returns {pos: [1], peaks: [2]} (or equivalent in other languages) """
+
+
+def pick_peaks(arr):
+    answer = {"pos": [], "peaks": []}
+    if not arr:
+        return answer
+
+    direction = ""
+    previous_number = arr[0]
+    previous_pos = 0
+    answer = {"pos": [], "peaks": []}
+
+    for pos, number in enumerate(arr[1:], 1):
+        if number > previous_number:
+            previous_number = number
+            previous_pos = pos
+            direction = "uphill"
+        elif number == previous_number:
```

Comment:
`for pos, number: `
`if previous< number and number >= next: peak`
wouldn't be a bit shorter?

Reply:
It's good idea, but there is a problem with "plateaus."
For example, [1,2,2,2,2] and [1,2,2,2,3] have no peaks.
I change code using your advice, but it needs one more loop. In spite of this
 new code looks more simply. But I think it requires more operations

Reply:
It's a valid. One more idea: find all local maximums as I suggested. 
Than we can find out all local minimums  previous>= num >=next
After that we can check that after each local max would be local min lesser than it - than it would be correct peak. Othervise (max = next min) it would be just plateau